benchmarks/babelstream/src/benchmarkTest.cpp

```diff
@@ -0,0 +1,434 @@
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+
+#include <catch2/benchmark/catch_benchmark.hpp>
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+
+/**
+ * Babelstream benchmarking example. Babelstream has 5 kernels. Add, Multiply, Copy, Triad and Dot.
+ * Babelstream is a memory-bound benchmark since the main operation in the kernels has high Code Balance (bytes/FLOP)
+ * value. For example c[i] = a[i] + b[i]; has 2 reads 1 writes and has one FLOP operation. For double precision each
+ * read-write is 8 bytes. Hence Code Balance (3*8 / 1) = 24 bytes/FLOP.
+ *
+ * Some implementations and the documents are accessible through https://github.com/UoB-HPC
+ */
+
+namespace
+{
+    constexpr auto arraySizeMain = 1024 * 1024;
+    // Mul and Triad kernel params
+    constexpr auto startScalar = 2.0f;
+    // DotKernel test workdiv params
+    constexpr auto blockThreadExtentMain = 1024;
+
+    //!
+    //! \brief FuzzyEqual compares two floating point or integral type values
+    //!
+    template<typename T>
+    bool FuzzyEqual(T a, T b)
+    {
+        if constexpr(std::is_floating_point_v<T>)
+        {
+            return std::fabs(a - b) < std::numeric_limits<T>::epsilon() * static_cast<T>(1000.0);
+        }
+        else if constexpr(std::is_integral_v<T>)
+        {
+            return a == b;
+        }
+        else
+        {
+            static_assert(
+                std::is_floating_point_v<T> || std::is_integral_v<T>,
+                "FuzzyEqual<T> is only supported for integral or floating-point types.");
+        }
+    }
+
+    std::string getCurrentTimestamp();
```

Comment:
Why the forward declaration?

Reply:
There was a warning at one compiler  "`warning: no previous declaration for ‘std::string getCurrentTimestamp()’ [-Wmissing-declarations]`" therefore i added a declaration. Alternatively, I can make the function `static` as well. Somehow one of the compilers forced for declaration.

Reply:
I made the function static. Like functions in some other tests: e.g. `alpaka/test/unit/mem/buf/src/BufTest.cpp`